migrations/20240202003133_better_dequeue_index.sql

```diff
@@ -0,0 +1,9 @@
+DROP INDEX idx_queue_scheduled_at;
+
+/*
+Partial index used for dequeuing from job_queue.
+
+Dequeue only looks at available jobs so a partial index serves us well.
+Moreover, dequeue sorts jobs by attempt and scheduled_at, which matches this index.
+*/
+CREATE INDEX idx_queue_dequeue_partial ON job_queue(queue, attempt, scheduled_at) WHERE status = 'available' :: job_status;
```

Comment:
```suggestion
CREATE INDEX CONCURRENTLY idx_queue_dequeue_partial ON job_queue(queue, attempt, scheduled_at) WHERE status = 'available' :: job_status;
```
Without `CONCURRENTLY`, creating an index on a table will lock it completely. cannot affort it on prod. See https://www.postgresql.org/docs/current/sql-createindex.html#SQL-CREATEINDEX-CONCURRENTLY

Reply:
By the way, `DROP INDEX` can also be done concurrently. I think it's worth it there too!